### PR TITLE
Extend valid resources for project-scoped role policy 

### PR DESCRIPTION
**Pull request by onematchfox**
Enables the ability to create project-role policies for the resources `clusters`, `applications`, `exec` and `logs`.

Notes: 
- Bumping the `argo-cd` go dependency required dropping support for `ksonnet` as per the [`2.4` release notes](https://argo-cd.readthedocs.io/en/latest/operator-manual/upgrading/2.3-2.4/#ksonnet-support-is-removed). Not sure quite how you want to handle this from the perspective of this provider given that it could definitely be seen as a breaking change. Let me know if anything is needed from my side.
- Replacement for #203 which went stale. 
- I have upgraded the default Kubernetes and ArgoCD versions that are installed as part of the tests (see commit messages for more details) - happy to drop these commits if you have any issues with them.

### CODE REVIEW

**Comment by onematchfox** [comment]
@oboukili is it just me or are the tests a bit flaky?``v2.1.10` passed on the [previous attempt](https://github.com/oboukili/terraform-provider-argocd/actions/runs/3151063899/jobs/5124563421) but failed on the [latest attempt](https://github.com/oboukili/terraform-provider-argocd/actions/runs/3151169779/jobs/5124794927) (for seemingly unrelated reasons). 

Will leave this PR as is for now. Let me know if there's anything I need to do about the tests (in addition to any review comments of course :smile:)

**Comment by onematchfox** [comment]
> Could you please fix the issue and publish the new version ?

I'm afraid there's not much I can do here -  I'm waiting on a review. I'm sure @oboukili will get to this when he has the time.

> PS : I am wondering if support of 2.1 is still needed ? (it not supported anymore according to documentation)

:thinking: this is a fair point - even by the providers own compatibility promise versions 2.1 and 2.2 are unsupported. So, I could drop the tests for these versions. But, I would rather wait for a maintainer's thoughts on the matter than make that call myself.

File: scripts/testacc_prepare_env.sh
Comment (oboukili) [inline scripts/testacc_prepare_env.sh:25]: Let's drop support for 2.1.x, as per the compatibility promise, and let's raise this to the latest 2.3.x.

File: argocd/resource_argocd_application.go
Comment (oboukili) [inline argocd/resource_argocd_application.go:52]: Could you add a schemaUpgrader test? If something goes wrong, we'll have to release hotfixes under pressure (already experienced that ;) )

**Review by oboukili — CHANGES_REQUESTED** [review]

**Comment by oboukili** [comment]
Hi there,
Apologies for the slow response time, indeed everybody is pretty busy these days. @onematchfox since you're already a long time contributor, would you mind if I added you as a repository maintainer?
Cheers and thanks,

File: scripts/testacc_prepare_env.sh
Comment (onematchfox) [reply scripts/testacc_prepare_env.sh]: Done.

File: argocd/resource_argocd_application.go
Comment (onematchfox) [reply argocd/resource_argocd_application.go:52]: Good catch :) Added as requested.

**Comment by onematchfox** [comment]
> Hi there, Apologies for the slow response time, indeed everybody is pretty busy these days. 

Thanks so much for the review. Have now completed the changes as requested. 

> @onematchfox since you're already a long time contributor, would you mind if I added you as a repository maintainer? Cheers and thanks,

I'm by no means an expert in Terraform providers :smile: but yes sure, I'd be more than happy to help out wherever possible if it would lighten your load.

File: argocd/utils.go
Comment (oboukili) [inline argocd/utils.go:100]: There is a comment to refactor this using a data structure `rbac.ResourceApplications` data structure maintained upstream, did you consider that instead?

File: scripts/testacc_prepare_env.sh
Comment (oboukili) [inline scripts/testacc_prepare_env.sh:25]:
```suggestion
echo "\n--- Install ArgoCD ${ARGOCD_VERSION:-v2.4.12}\n"
```

File: scripts/testacc_prepare_env.sh
Comment (oboukili) [inline scripts/testacc_prepare_env.sh:26]:
```suggestion
curl https://raw.githubusercontent.com/argoproj/argo-cd/${ARGOCD_VERSION:-v2.4.12}/manifests/install.yaml > manifests/install/argocd.yml &&
```

**Review by oboukili — CHANGES_REQUESTED** [review]

**Comment by oboukili** [comment]
> I'm by no means an expert in Terraform providers 😄 but yes sure, I'd be more than happy to help out wherever possible if it would lighten your load.

No worries, I'll set this up. Cheers!

File: argocd/utils.go
Comment (onematchfox) [reply argocd/utils.go]: I hadn't... Glossed over it because of the related comment for `validActions` at https://github.com/argoproj/argo-cd/blob/c99669e088b5f25c8ce8faff6df25797a8beb5ba/pkg/apis/application/v1alpha1/types.go#L1527

But now that you mention it, this module won't have any "import cycle" issues. So I have gone ahead and adjusted to reference `rbacpolicy` instead of hard-coding the strings (handled actions as well while I was at it)

**Comment by onematchfox** [comment]
@oboukili Changes now made. Acceptance Tests on `v2.4.12` failed initially for seemingly unrelated reasons so I have relaunched them.

**Comment by oboukili** [comment]
Awesome, thanks :)
Since there is a major breaking change, we are going to increment the major version counter.

**Comment by VladoPortos** [comment]
Guys how do I fix this?

```
Error decoding "argocd_application.gitlab" from previous state: unsupported attribute "ksonnet"
```

Just remove it from state file ?

**Comment by onematchfox** [comment]
Were/are you using `ksonnet`? 

> Just remove it from state file ?

Yes, that should do the trick.

**Comment by VladoPortos** [comment]
Worked, thanks @onematchfox

**Pull request merged**
